tests/app/views/handlers/test_submission_handler.py

```diff
@@ -27,26 +27,79 @@ def setUp(self):
             ru_ref="ru_ref",
             case_id="0123456789000000",
         )
-        self.session_store = SessionStore("user_ik", "pepper", "eq_session_id")
         self.expires_at = datetime.now(tzutc()) + timedelta(seconds=5)
-
-    def test_submission_language_code_in_payload(self):
-        session_store = self.session_store.create(
+        self.session_store = SessionStore("user_ik", "pepper", "eq_session_id").create(
             "eq_session_id", "user_id", self.session_data, self.expires_at
         )
-        storage = Mock()
-        storage.get_user_data = Mock(return_value=("{}", 1))
 
+    def test_submission_language_code_in_payload(self):
         with patch(
             "app.views.handlers.submission.get_session_store",
-            return_value=session_store,
+            return_value=self.session_store,
         ):
             with patch(
                 "app.views.handlers.submission.convert_answers", return_value={}
             ):
                 submission_handler = SubmissionHandler(
-                    QuestionnaireSchema({}), QuestionnaireStore(storage), {}
+                    QuestionnaireSchema({}), self.questionnaire_store_mock(), {}
                 )
                 assert (
                     submission_handler.get_payload()["submission_language_code"] == "cy"
                 )
+
+    def test_questionnaire_store_delete_called(self):
+        questionnaire_store = self.questionnaire_store_mock()
+        self.questionnaire_store_delete_mock(questionnaire_store)
```

Comment:
This method doesn't add much. I think it should just be inline for clarity or made it part of `self.questionnaire_store_mock()`.